### ISSUE TITLE
Why does Ciaos dataset with PMF get RMSE 2.823082?

### ISSUE THREAD

**Issue by asoboy** [guoguibing/librec#11]
Hi, glad to encounter such a nice java RS project! I just have met some problems with the 1.2ver. code or dataset, and feel difficult to find out the cause.

I tried Movielens-1M and Ciao dataset, both have the result of PMF on example page, my results under the Configurations from the example page are following:
Movielens-1M:
		MAE		RMSE
PMF		0.746997	0.926199 (nearly the same with the comparison data [0.747	0.926] in the example page)

Configurations:
PMF:
factors=10, reg=0.1, learn.rate=50, momentum=0.8, max.iter=100

Ciaos:
		MAE		RMSE
PMF		2.498656	2.823082 (while the exmaple gives [0.822 1.078] under the same configuration)

Configurations:
PMF:
factors=10, reg=0.4, learn.rate=50, momentum=0.8, max.iter=200

What's more, Ciaos dataset with SVD++ gets 0.702264	0.947451 which is much better than the PMF score, I cannot find the reason, could you help me with this result, thank you very much!

Configurations:
SVD++:
factors=10, reg=0.1, learn.rate=0.01, max.iter=100

**Comment by guoguibing** [comment]
Better performance may be obtained after tuning the moment larger.

**Comment by asoboy** [comment]
Thanks for your commenting!! 

The Ciao dataset has 35835 ratings and is downloaded from here (http://www.public.asu.edu/~jtang20/datasetcode/ciao_with_rating_timestamp_txt.zip).

I tested with momentum from 0.1 to 0.9 (the 1.0 will cause crach due to the Loss grows into infinity), and the result is here:

learning Rate = 50, regP = regQ = 0.1, K = 10, iters = 100
>			MAE	 	RMSE 	Momentum
>PMF		2.484504	2.807264	0.1
>PMF		2.516064	2.832653	0.2
>PMF		2.471362	2.803306	0.3
>PMF		2.452357	2.783117	0.4
>PMF		2.464851	2.797973	0.5
>PMF		2.582281	2.890350	0.6
>PMF		2.562090	2.872376	0.7
>PMF		2.591306	2.898688	0.8
>PMF		2.553969	2.868095	0.9

I am not sure about whether it is the dataset's flaw or not. Could you help run with this dataset and return the confiuguration and result of PMF, Thank you!!

**Comment by guoguibing** [comment]
moment is not restricted in [0, 1], try larger values like 10, 50, etc. 

I do not have spare time to run your tests.

**Comment by guoguibing** [comment]
BTW, one of my collaborators mentioned that you may want to check the format of ciao data set.

**Comment by 466152112** [comment]
When I use the v1.2, I find the item or user which no include in train set still be calulated in  evaluation stage. So please check if is this issue. We will repair this issue as soon as possible.
